File: 03410-SAMER08F-Feynman/SAMER08F.lua
--[[
SAMER08F
]]

-- GLOBALES -----------------------------------------------------------------------------
main, resolver = nil
-- FIN GLOBALES -------------------------------------------------------------------------

function main()
  local n, lin
  
  while (true) do
    lin = io.read()
    n = tonumber(lin)
    if (n == 0) then break; end

    print(resolver(n))
  end
end

function resolver(n)
  local total, cpl
  
  total = 0
  for tam = 1, n do
    cpl = n - tam + 1   -- cuadrados solapados por lado
    total = total + (cpl * cpl)
  end
  return total
end

main()
os.exit()

File: 03410-SAMER08F-Feynman/SAMER08F.c
#include <stdio.h>
#include <stdlib.h>

int resolver(int n);

int main(void) {
  int n;
  char lin[255];
  
  while (1) {
    scanf("%s\n", lin);
    n = atoi(lin);
    if (n == 0) break;
    
    printf("%d\n", resolver(n));
  }
  exit(0);
}

int resolver(int n) {
  int tam, total, cpl;
  
  total = 0;
  for (tam = 1; tam <= n; tam++) {
    cpl = n - tam + 1;
    total += (cpl * cpl);
  }
  return total;
}
